app/controllers/documents/requested_documents_later_controller.rb

```diff
@@ -87,11 +87,16 @@ def current_session_or_home
     end
 
     def handle_session
-      return if session[:documents_request_id].present?
+      return if session[:documents_request_id].present? &&
```

Comment:
non-blocking: consider moving this into a function name that describes what's happening